PowerPointLabs/Test/FunctionalTest/TimerLabTest.cs

```diff
@@ -0,0 +1,156 @@
+            int expectedColor = PpOperations.SelectShape(TimerLineMarkerGroup)[1].Line.ForeColor.RGB;
+
+            PpOperations.SelectSlide(OriginalSlideNo);
+            var lineMarkerGroup = PpOperations.SelectShape(TimerLineMarkerGroup);
+            lineMarkerGroup.Line.ForeColor.RGB = expectedColor;
+            var timeMarkerGroup = PpOperations.SelectShape(TimerTimeMarkerGroup);
+            timeMarkerGroup.Delete();
+
+            MessageBoxUtil.ExpectMessageBoxWillPopUp("Error",
+              "Only one timer allowed per slide.", timerLab.ClickCreateButton, "Ok");
+            AssertIsSame(OriginalSlideNo, ChangeLineColorAndRecreateTimeMarkerSlideNo);
+        }
+
+        private void TestEditTimerDuration(ITimerLabController timerLab)
+        {
+            PpOperations.SelectSlide(OriginalSlideNo);
+            timerLab.SetDurationTextBoxValue(0.07);
+            AssertIsSame(OriginalSlideNo, ChangeDurationSlideNo);
+        }
+
+        private void TestEditTextColorAndDeleteSlider(ITimerLabController timerLab)
+        {
+            PpOperations.SelectSlide(ChangeTextColorAndRecreateSliderSlideNo);
+            int expectedColor = PpOperations.SelectShape(TimerTimeMarkerGroup)[1].TextFrame.TextRange.Font.Color.RGB;
+
+            PpOperations.SelectSlide(OriginalSlideNo);
+            var timeMarkerGroup = PpOperations.SelectShape(TimerTimeMarkerGroup);
+            timeMarkerGroup.TextFrame.TextRange.Font.Color.RGB = expectedColor;
+            List<string> sliderComponentNames = new List<string> { TimerSliderHead, TimerSliderBody };
+            var sliderComponents = PpOperations.SelectShapes(sliderComponentNames);
+            sliderComponents.Delete();
+
+            MessageBoxUtil.ExpectMessageBoxWillPopUp("Error",
+              "Only one timer allowed per slide.", timerLab.ClickCreateButton, "Ok");
+            AssertIsSame(OriginalSlideNo, ChangeTextColorAndRecreateSliderSlideNo);
+        }
+
+        private void TestEditDurationAndWidth(ITimerLabController timerLab)
+        {
+            PpOperations.SelectSlide(OriginalSlideNo);
+            timerLab.SetDurationTextBoxValue(4.56);
+            timerLab.SetWidthSliderValue(654);
+
+            AssertIsSame(OriginalSlideNo, ChangeDurationAndWidthSlideNo);
+        }
+
+        private void TestInvalidDuration(ITimerLabController timerLab)
+        {
+            PpOperations.SelectSlide(OriginalSlideNo);
+            timerLab.SetDurationTextBoxValue(5.67);
+
+            AssertIsSame(OriginalSlideNo, DurationInvalidSlideNo);
+        }
+
+        private void AssertIsSame(int actualSlideNo, int expectedSlideNo)
+        {
+            var actualSlide = PpOperations.SelectSlide(actualSlideNo);
+            var expectedSlide = PpOperations.SelectSlide(expectedSlideNo);
+
+            SlideUtil.IsSameShapes(expectedSlide, actualSlide);
+            SlideUtil.IsSameAnimations(expectedSlide, actualSlide);
```

Comment:
Oh ok i missed that, the method needs to be updated to check for the motion path and duration though.